src/main/java/work/xeltica/craft/core/modules/coreProtectGuiTest/CoreProtectGuiTestApp.kt

```diff
@@ -0,0 +1,270 @@
+package work.xeltica.craft.core.modules.coreProtectGuiTest
+
+import org.bukkit.Bukkit
+import org.bukkit.Material
+import org.bukkit.entity.Player
+import work.xeltica.craft.core.xphone.apps.AppBase
+
+/**
+ * CoreProtectのコマンドを良い感じにXPhoneから叩くようにするアプリ
+ */
+class CoreProtectGuiTestApp : AppBase() {
+
+    /**
+     * コマンドの値として入れる値
+     */
+    private val coreProtectCommand by lazy { CoreProtectCommand() }
+
+    /**
+     * [CoreProtectGuiTestModule]
+     *
+     * NOTE: アプリとモジュールでのやりとりがかなり多いのでインスタンス生成処理の削減のためにメンバ変数として宣言する
+     */
+    private val module by lazy { CoreProtectGuiTestModule }
+    private val menuList by lazy { CoreProtectGuiTestModule.GetMenuList() }
+
+    /**
+     * 範囲を指定するときにつかう
+     */
+    private lateinit var firstDateTime: Pair<Int, String>
+    private lateinit var secondDateTime: Pair<Int, String>
+
+    var commandFirst: String = "co"
+
+    override fun getName(player: Player): String {
+        return "CoreProtectGUIテストアプリ"
+    }
+
+    override fun getIcon(player: Player): Material {
+        return Material.ENDER_EYE
+    }
+
+    override fun onLaunch(player: Player) {
+        module.start(player)
+    }
+
+    /**
+     * 1度目に取得した日時と2度目に取得した日時を比較して
+     * [CoreProtectCommand.date]に格納する文字列を決定するやつ
+     */
+    private fun checkDuringTime() {
+        val firstDate = firstDateTime.first.toLong() * convertUnit(firstDateTime.second)
+        val lastDate = secondDateTime.first.toLong() * convertUnit(secondDateTime.second)
+
+        if (firstDate < lastDate) {
+            coreProtectCommand.date = (firstDateTime.first.toString() + firstDateTime.second) + "-" + (secondDateTime.first.toString() + secondDateTime.second)
+            Bukkit.getLogger().info("FirstDate more than")
+        } else if (firstDate > lastDate) {
+            coreProtectCommand.date = (secondDateTime.first.toString() + secondDateTime.second) + "-" + (firstDateTime.first.toString() + firstDateTime.second)
+            Bukkit.getLogger().info("LastDate more than")
+        } else {
+            // ここに到達する場合は同値である場合なので、onTimeUnitMenuClick と同じ経路になるような形で
+            // コマンドの文字列を確定させる
+            coreProtectCommand.date = firstDateTime.first.toString() + firstDateTime.second
+        }
+        Bukkit.getLogger().info(coreProtectCommand.date)
+    }
+
+    /**
+     * 単位を秒数に変換する
+     *
+     * @param unit 単位
+     * @return 単位を秒数に変換した数値。値がでかくなるので[Long]で変換
+     *
+     * NOTE: 合理的な処理が思いつかなかった
+     */
+    private fun convertUnit(unit: String): Long {
+        return when (unit) {
+            "s" -> {
+                1
+            }
+
+            "m" -> {
+                60
+            }
+
+            "h" -> {
+                3600
+            }
+
+            "d" -> {
+                86400
+            }
+
+            "w" -> {
+                604800
+            }
+
+            else -> {
+                0
+            }
```

Comment:
* whenの中身は単純な式を帰す場合、 `{}` を省略できます
* もっと丁寧に書くなら、マジックナンバーみが増さないように途中式にしてしまったり、変なunitは弾くよう例外処理をするほうが良いかも

```suggestion
            "s" -> 1
            "m" -> 1 * 60
            "h" -> 1 * 60 * 60
            "d" -> 1 * 60 * 60 * 24
            "w" -> 1 * 60 * 60 * 24 * 7
            else -> throw IllegalArgumentException("Invalid unit \"${unit}\"")
```